docs(profiling): Update python onboarding to use continuous profiling (#13505)
getsentry/sentry-docs: docs/platforms/python/configuration/options.mdx

## Profiling Options

<SdkOption name="profile_lifecycle" type='str' defaultValue='manual'>

> Can be either `manual` or `trace`. The lifecycle chosen determines how the profile will be run.

Reviewer: ```suggestion
Determines how the profiler runs, accepts two values: `manual` or `trace`. 
```

> In `manual` mode, the profiling data collection can be managed via calls to `sentry_sdk.profiler.start_profiler` and `sentry_sdk.profiler.stop_profiler`. You are entirely in control of when the profiler runs.

Reviewer: ```suggestion
In `manual` mode, you control when the profiler collects data by explicitly calling `sentry_sdk.profiler.start_profiler` and `sentry_sdk.profiler.stop_profiler`. You are entirely in control of when the profiler runs.
```

> In `trace` mode, the profiler manages its own start and stop calls, which are based on spans: the profiler continues to run while there is at least one active span, and stops when there are no active spans.

Reviewer: ```suggestion
In `trace` mode, the profiler starts and stops automatically based on active spans: the profiler continues to run while there is at least one active span, and stops when there are no active spans.
```
